pkg/controller/replicaset/replica_set.go

```diff
@@ -176,8 +176,8 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 
 // getPodReplicaSet returns the replica set managing the given pod.
 // TODO: Surface that we are ignoring multiple replica sets for a single pod.
-// TODO: use ownerReference.Controller to determine if the rs controls the pod.
 func (rsc *ReplicaSetController) getPodReplicaSet(pod *v1.Pod) *extensions.ReplicaSet {
+	controllerRef := controller.GetControllerOf(pod.ObjectMeta)
```

Comment:
We shouldn't need to list all replica sets and search through the list, we should simply do a `rsc.rsLister.Get` instead of running `rsc.rsLister.GetPodReplicaSets(pod)`